src/Microsoft.Extensions.Http.Polly/DependencyInjection/PollyHttpClientBuilderExtensions.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using Microsoft.Extensions.Http;
+using Polly;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class PollyHttpClientBuilderExtensions
+    {
+        public static IHttpClientBuilder AddPolicyHandler(this IHttpClientBuilder builder, IAsyncPolicy policy)
+        {
+            //builder.AddHttpMessageHandler(() => new PolicyHttpMessageHandler(createPolicy()));
+            return builder;
+        }
+
+        public static IHttpClientBuilder AddPolicyHandler(this IHttpClientBuilder builder, IAsyncPolicy<HttpResponseMessage> policy)
+        {
+            //builder.AddHttpMessageHandler(() => new PolicyHttpMessageHandler(createPolicy()));
+            return builder;
+        }
+
+        public static IHttpClientBuilder AddExceptionPolicyHandler(this IHttpClientBuilder builder, Func<PolicyBuilder, IAsyncPolicy> createPolicy)
+        {
+            var policyBuilder = Policy.Handle<HttpRequestException>();
+            var policy = createPolicy(policyBuilder);
+
+            //builder.AddHttpMessageHandler(() => new PolicyHttpMessageHandler(createPolicy()));
+            return builder;
+        }
+
+        public static IHttpClientBuilder AddExceptionPolicyHandler(this IHttpClientBuilder builder, Func<PolicyBuilder, IAsyncPolicy<HttpResponseMessage>> createPolicy)
+        {
+            //builder.AddHttpMessageHandler(() => new PolicyHttpMessageHandler(createPolicy()));
+            return builder;
+        }
+
+        public static IHttpClientBuilder AddServerErrorPolicyHandler(
+            this IHttpClientBuilder builder, 
+            Func<PolicyBuilder<HttpResponseMessage>, IAsyncPolicy<HttpResponseMessage>> createPolicy)
+        {
+            var policyBuilder = Policy.Handle<HttpRequestException>().OrResult<HttpResponseMessage>(response =>
+            {
+                return response.StatusCode >= HttpStatusCode.InternalServerError;
+            });
+            var policy = createPolicy(policyBuilder);
+
+            //builder.AddHttpMessageHandler(() => new PolicyHttpMessageHandler(createPolicy()));
+            return builder;
+        }
+
```

Comment:
Polly policies divide into reactive policies (Retry; CircuitBreaker; Fallback), and proactive and preemptive policies (Cache; Bulkhead; Timeout).  The reactive policies are the ones which flow from `PolicyBuilder`, so the choice of configuration syntax here with variants around `Func<PolicyBuilder..., IAsyncPolicy...>` focuses on (only) the Retry, CircuitBreaker and Fallback policies.

I'm drawing this out in case it wasn't intended, but it seems reasonable, as Retry and CircuitBreaker seem likely majority use cases for Polly with `HttpClientFactory`.  And for those who want to go further, the more advanced overloads (eg `AddPolicyHandler(this IHttpClientBuilder builder, IAsyncPolicy policy)` line 14) do provide.

Reply:
Thanks for the feedback. This will be good info for writing the docs.